server/server.go

```diff
@@ -422,3 +406,34 @@ func (s *Server) StartExitMonitor() {
 	}
 	<-done
 }
+
+// startStatusUpdateWorker accepts jobs in the form of container IDs whose
+// status is updated by the function.
+func (s *Server) startStatusUpdateWorker(workerID int, containerJobs <-chan string) {
+	for containerID := range containerJobs {
+		logrus.Debugf("StatusUpdateWorker %v processing %v", workerID, containerID)
+		c := s.GetContainer(containerID)
+		if c != nil {
+			logrus.Debugf("container exited and found: %v", containerID)
+			err := s.Runtime().UpdateStatus(c)
+			if err != nil {
+				logrus.Warnf("Failed to update container status %s: %v", c, err)
+			} else {
```

Comment:
You could clean this up a bit, by using continues  on error, and just calling 
 +					s.ContainerStateToDisk(c) 
at the bottom of the loop.